### PR TITLE
VIH-9562 support dom1 login

### PR DESCRIPTION
**Pull request by shaed-parkar**
### JIRA link (if applicable) ###

VIH-9562

### Change description ###

* Add DOM1 AD Scheme
* Add a middleware to get a user's roles from Bookings API
* Use the `DOM1` feature flag to determine whether or not to add DOM1 as an available provider
* Use the new `multiIdpSelection` feature flag to determine whether or not to direct user to IDP selection instead of `EJUDFeature` since Ejud may not be enabled but DOM1 may
* Update IDP Selection component to update accordingly when features toggle

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts [hunk 1]
```diff
@@ -4,32 +4,73 @@ import { Logger } from 'src/app/services/logging/logger-base';
...
+
+        this.ldService.getFlag(FEATURE_FLAGS.dom1SignIn).subscribe(featureEnabled => {
+            if (featureEnabled) {
+                this.identityProviders[IdpProviders.dom1] = {
+                    url: '/' + pageUrls.Login
+                };
+                this.updateProviderNames();
+            }
+        });
+        // this.ldService.flagChange.subscribe(value => {
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts:43]: some commented out code here - can we remove it if it's not needed?

File: VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts [hunk 2]
```diff
@@ -4,32 +4,73 @@ import { Logger } from 'src/app/services/logging/logger-base';
...
+        this.updateProviderNames();
     }
 
     showError(): boolean {
         return this.submitted && !this.selectedProvider;
     }
 
+    updateProviderNames(): void {
+        this.identityProvidernames = Object.keys(this.identityProviders).reverse();
+        // return Object.keys(this.identityProviders).reverse();
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts:69]: some commented out code here - could we remove it

**Review by oliver-scott — COMMENTED** [review]
the code looks ok to me, just some minor comments.

looks like there are some failing tests though, and you might need to change the PR title

File: VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/security/idp-selection/idp-selection.component.ts]: done, lost a few commits along the way. the component is correct now

**Review by komaldabhi1509 — APPROVED** [review]

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**